mirgecom/timesteppers/lsrk.py

```diff
@@ -0,0 +1,131 @@
+from mirgecom.timesteppers.base import TimestepperBase
+import numpy as np
+
+
+__all__ = ("LSRKEuler", "LSRK54CarpenterKennedy",
+           "LSRK144NiegemannDiehlBusch")
+
+
+class LowStorageRungeKutta(TimestepperBase):
+    """An amazingly descriptive docstring."""
+
+    def __init__(self, A, B, C):
+        """An amazingly descriptive docstring."""
+
+        super(LowStorageRungeKutta, self).__init__()
+        self.A = A
+        self.B = B
+        self.C = C
+        self.nstages = len(B)
+
+    def step(self, state, t, dt, rhs):
+        """Take one step using a low-storage Runge Kutta method."""
+
+        k = 0.0
+        for i in range(self.nstages):
+            k = self.A[i]*k + dt*rhs(t + self.C[i]*dt, state)
+            state += self.B[i]*k
+
+        return state
```

Comment:
Seems like you might be able to achieve this without resorting to a class hierarchy, by doing something like:

```python
@dataclass(frozen=True)
class LSRK4Coefs:
    A: np.ndarray
    B: np.ndarray
    C: np.ndarray


def lsrk4_step(coefs, state, t, dt, rhs):
     k = 0.0
     for i in range(len(coefs.A)):
         k = coefs.A[i]*k + dt*rhs(t + coefs.C[i]*dt, state)
         state += coefs.B[i]*k
     return state
```

```python
LSRK54CarpenterKennedyCoefs = LSRK4Coefs(
    A=[...],
    B=[...],
    C=[...])


lsrk4_carpenter_kennedy_step = partial(lsrk4_step, LSRK54CarpenterKennedyCoefs)
```
Thoughts?

Reply:
Could even do:

```python
lsrk54_carpenter_kennedy_step = partial(lsrk_step, LSRKCoefs(
    A=[...],
    B=[...],
    C=[...])
```
🙂

Reply:
Then again, if you're going to do that maybe you don't even need the dataclass:

```python
def lsrk_step(A, B, C, state, t, dt, rhs):
    ...
```

```python
lsrk54_carpenter_kennedy_step = partial(lsrk_step, 
    A=[...],
    B=[...],
    C=[...])
```

I'm rambling here, sorry. 🤦

Reply:
@majosm I like the pythonics :smiley: I will try out your suggestions and see how it "feels" and "looks", then update! Out of curiosity, are there some subtle disadvantages to using a class hierarchy that I may be overlooking? Perhaps it's a bit unnecessary since the steppers are relatively simple algorithms

Reply:
Mainly just extra boilerplate code I think. Classes make sense when you have a more complicated interface, but for something that is effectively just a function, you might as well make it one. 🙂

Reply:
One potential downside I just noticed to what I suggested above: we won't get a unique docstring for each integrator function. Maybe it would be better to do (with the dataclass again):

```python
LSRK54CarpenterKennedyCoefs = LSRKCoefs(
    A=[...],
    B=[...],
    C=[...])


def lsrk54_carpenter_kennedy_step(state, t, dt, rhs):
    """Blah blah blah."""
    return lsrk_step(LSRK54CarpenterKennedyCoefs, state, t, dt, rhs)
```

Reply:
@majosm I decided to go with your dataclass suggestion. With regards to the docstring, would
```python
foo = partial(func, args)
foo.__doc__ = ...
```
Be a viable alternative? I just now saw the discussion after I already pushed some changes. I don't mind updating the function definitions to what you're suggesting here if this is preferrable!

Reply:
Hmm, maybe -- not sure. @inducer?

Reply:
I've decided to go with your suggestion anyways because I think it's more clear!